### ISSUE TITLE
Wrong warning about unused import

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust#50087]
Hi.
In my crate relm, I have [an example](https://github.com/antoyo/relm/tree/master/examples/buttons-derive) that shows warnings about unused imports:
```
warning: unused imports: `ButtonExt`, `LabelExt`, `OrientableExt`, `WidgetExt`
  --> src/main.rs:29:5
   |
29 |     ButtonExt,
   |     ^^^^^^^^^
30 |     Inhibit,
31 |     LabelExt,
   |     ^^^^^^^^
32 |     OrientableExt,
   |     ^^^^^^^^^^^^^
33 |     WidgetExt,
   |     ^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default
```
But if I comment them, I get compile errors because these import are actually needed.
Thanks to fix the issue.

**Comment by ExpHP** [comment]
Looks like #45268. My guess is that one of those macros generates a `mod` and uses `use super::*` inside of it?

**Comment by antoyo** [comment]
Yes, it does `use super::*`.

**Comment by ExpHP** [comment]
Okay, for the purposes of keeping things tidy I think this issue should be closed in favor of the other

**Comment by antoyo** [comment]
You're right.
Closed in favor of #45268.
